expression/builtin_control.go

```diff
@@ -365,7 +365,6 @@ func (c *ifNullFunctionClass) getFunction(args []Expression, ctx context.Context
 		evalTps = tpString
 		if !types.IsBinaryStr(tp0) && !types.IsBinaryStr(tp1) {
```

Comment:
https://github.com/pingcap/tidb/blob/master/expression/builtin_control.go#L147 seems wrong and the rewritten version seem right.
I test 'IF' && 'IFNULL' in mysql. the following `c` is of `char(10)` type , and `f` is of `binary(1)` type.  The results are as follows:
```
mysql> select if(1>2,c,f) from t;
Field   1:  `if(1>2,c,f)`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       STRING
Collation:  binary (63)
Length:     10
Max_length: 0
Decimals:   31
Flags:      BINARY 


0 rows in set (0.00 sec)

mysql> select ifnull(c,f) from t;
Field   1:  `ifnull(c,f)`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       STRING
Collation:  binary (63)
Length:     10
Max_length: 0
Decimals:   31
Flags:      BINARY 


0 rows in set (0.00 sec)

```